edi_sale_utm_oca/models/sale_order.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _name = "sale.order"
+    _inherit = [
+        "sale.order",
+        "edi.utm.consumer.mixin",
+    ]
+
+    def _set_source_edi(self, vals):
+        if vals.get("origin_exchange_record_id"):
+            exchange_record = self.env["edi.exchange.record"].browse(
+                vals["origin_exchange_record_id"]
+            )
+
+            exchange_type = exchange_record.type_id
+
+            conditions = [
+                exchange_type,
+                exchange_type.edi_source_id,
+                "edi_source_id" not in vals,
+            ]
+
+            if all(conditions):
+                vals["edi_source_id"] = exchange_type.edi_source_id.id
+
+        return vals
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            self._set_source_edi(vals)
+        return super().create(vals_list)
```

Comment:
Why you don't use `_edi_set_source` on `edi_utm_oca` 

And please put this one in `def _edi_set_source` of `edi_utm_oca`

```
if vals.get("origin_exchange_record_id"):
            exchange_record = self.env["edi.exchange.record"].browse(
                vals["origin_exchange_record_id"]
            )

            exchange_type = exchange_record.type_id

            conditions = [
                exchange_type,
                exchange_type.edi_source_id,
                "edi_source_id" not in vals,
            ]

            if all(conditions):
                vals["edi_source_id"] = exchange_type.edi_source_id.id

Reply:
sorry, please ignore the block, just need to re-use `_edi_update_source`